### PR TITLE
fix(table): Export handles newlines & expand rows.

### PR DESCRIPTION
**Pull request by LucasHocker**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Type of PR

Bugfix (non-breaking change which fixes an issue) 

#### Checklist

- [ X] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ X] Lint and unit tests pass locally with my changes
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/barista-components/table/src/table.ts [hunk 1]
```diff
@@ -549,11 +549,13 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
...
         ).rootNodes[0];
-        const cells = rowEl.childNodes;
+        const cells = rowEl.querySelectorAll('.dt-cell');
         for (let j = 0; j < cells.length; j++) {
           //if excluded, skip
           if (!keys.includes(columns[j])) continue;
-          const txt = cells[j].innerText;
+          let txt = cells[j].innerText;
+          //if an expandable cell, treat as blank
+          if (cells[j].matches('.dt-expandable-cell')) txt = '';
```
Comment (ffriedl89) [inline libs/barista-components/table/src/table.ts:558]:
```
let txt = cells[j].classList.contains('.dt-expandable-cell') ? '' : cells[j].innerText;
```

File: libs/barista-components/table/src/table.ts [hunk 2]
```diff
@@ -583,6 +585,7 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
   ): void {
     switch (typeof val) {
       case 'string':
+        val = val.replace(/[\n\r\t]/g, ' ');
```
Comment (ffriedl89) [inline libs/barista-components/table/src/table.ts:588]: Please add a comment what you are replacing here with this regex. I am really bad at regex :D

**Review by ffriedl89 — CHANGES_REQUESTED** [review]

File: libs/examples/src/table/table-expandable-rows-example/table-expandable-rows-example.html [hunk 3]
```diff
@@ -1,4 +1,8 @@
-<dt-table [dataSource]="dataSource" [multiExpand]="multiExpand">
+<dt-table
+  [dataSource]="dataSource"
+  [multiExpand]="multiExpand"
+  [showExportButton]="true"
```
Comment (Sherif-Elhefnawy) [inline libs/examples/src/table/table-expandable-rows-example/table-expandable-rows-example.html:4]: Please don't add export here. Either create a new example for exporting + expandable row or update one of the export stories to have the expandable rows

**Review by Sherif-Elhefnawy — CHANGES_REQUESTED** [review]

**Review by ffriedl89 — APPROVED** [review]

**Review by Sherif-Elhefnawy — APPROVED** [review]

**Comment by Sherif-Elhefnawy** [comment]
Cherry picked into:
10.7.x: 4a06d1add9869a7f437430a274e656ea4ee547ce

**Pull request merged**